packages/firestore/src/util/input_validation.ts

```diff
@@ -132,14 +132,10 @@ export function valueDescription(input: unknown): string {
   }
 }
 
-/** Hacky method to try to get the constructor name for an object. */
+/** try to get the constructor name for an object. */
 export function tryGetCustomObjectType(input: object): string | null {
   if (input.constructor) {
-    const funcNameRegex = /function\s+([^\s(]+)\s*\(/;
-    const results = funcNameRegex.exec(input.constructor.toString());
-    if (results && results.length > 1) {
-      return results[1];
-    }
+    return input.constructor.name;
```

Comment:
@schmidt-sebastian  I changed the compilation target to es2017 for testing, which preserves class definitions and revealed that this function doesn't work with classes. `constructor.name` is available everywhere except IE, which we don't need to worry about anymore. - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name#browser_compatibility